components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth/action/PreIssueAccessTokenRequestBuilderTest.java

```diff
@@ -0,0 +1,417 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+
+/**
+ * Unit test class for PreIssueAccessTokenRequestBuilder class.
+ */
+public class PreIssueAccessTokenRequestBuilderTest {
+
+    private PreIssueAccessTokenRequestBuilder preIssueAccessTokenRequestBuilder;
+
+    private MockedStatic<ClaimHandlerUtil> claimHandlerUtilMockedStatic;
+    private MockedStatic<OAuthServerConfiguration> oAuthServerConfiguration;
+    private MockedStatic<IdentityTenantUtil> identityTenantUtilMockedStatic;
+    private MockedStatic<OAuth2Util> oAuth2UtilMockedStatic;
+
+    private static final String CLIENT_ID_TEST = "test-client-id";
+    private static final String CLIENT_SECRET_TEST = "test-client-secret";
+    private static final String GRANT_TYPE_TEST = "password";
+    private static final String TENANT_DOMAIN_TEST = "carbon.super";
+    private static final int TENANT_ID_TEST = 1234;
+    private static final String USER_ID_TEST = "user-123";
+    private static final String USERNAME_TEST = "testUser";
+    private static final String PASSWORD_TEST = "test@123";
+    private static final String USER_STORE_TEST = "PRIMARY";
+    private static final String TEST_URL = "https://test.com/oauth2/token";
+    private static final String AUDIENCE_TEST = "audience1";
+
+    @BeforeClass
+    public void setUp() {
+
+        preIssueAccessTokenRequestBuilder = new PreIssueAccessTokenRequestBuilder();
+
+        OAuthServerConfiguration mockOAuthServerConfiguration = mock(OAuthServerConfiguration.class);
+        oAuthServerConfiguration = mockStatic(OAuthServerConfiguration.class);
+        oAuthServerConfiguration.when(OAuthServerConfiguration::getInstance).thenReturn(mockOAuthServerConfiguration);
+        oAuth2UtilMockedStatic = mockStatic(OAuth2Util.class);
+        OAuthAppDO oAuthAppDO = mock(OAuthAppDO.class);
+        oAuth2UtilMockedStatic.when(() -> OAuth2Util.getAppInformationByClientId(CLIENT_ID_TEST, TENANT_DOMAIN_TEST))
+                .thenReturn(oAuthAppDO);
+        oAuth2UtilMockedStatic.when(() -> OAuth2Util.getIdTokenIssuer(TENANT_DOMAIN_TEST)).thenReturn(TEST_URL);
+        doReturn(new String[]{AUDIENCE_TEST}).when(oAuthAppDO).getAudiences();
+        doReturn(CLIENT_ID_TEST).when(oAuthAppDO).getOauthConsumerKey();
+        oAuth2UtilMockedStatic.when(() -> OAuth2Util.getOIDCAudience(CLIENT_ID_TEST, oAuthAppDO)).
+                thenReturn(new LinkedList<>(Collections.singleton(AUDIENCE_TEST)));
+        oAuth2UtilMockedStatic.when(OAuth2Util::isPairwiseSubEnabledForAccessTokens).thenReturn(false);
+        AuthenticatedUser authenticatedUser = mock(AuthenticatedUser.class);
+        doReturn(CLIENT_ID_TEST).when(authenticatedUser).getAuthenticatedSubjectIdentifier();
```

Comment:
why do we return a client id for authenticated user. IMO, this should be USER_ID_TEST

Reply:
This is updated with ef43ee44e99c969f75d05fe1641cb112f816a7a4